book/src/user/shielded-scan-grpc-server.md

```diff
@@ -0,0 +1,143 @@
+# Zebra Shielded Scanning gRPC Server
+
+## Get Started
+
+### Setup
+
+After setting up Zebra Shielded Scanning, add a `listen_addr` field to the shielded-scan configuration:
+
+```toml
+[shielded_scan]
+listen_addr = "127.0.0.1:8231"
+```
+
+Then, run `zebrad` to start the scan gRPC server.
+
+Making requests to the server will also require a gRPC client, the examples here use `grpcurl`, though any gRPC client should work.
+
+[See installation instructions for `grpcurl` here](https://github.com/fullstorydev/grpcurl?tab=readme-ov-file#installation).
+
+The types can be accessed through the `zebra-grpc` crate's root `scanner` module for clients in a Rust environment, and the [`scanner.proto` file here](https://github.com/ZcashFoundation/zebra/blob/main/zebra-grpc/proto/scanner.proto) can be used to build types in other environments.
+
+### Usage
+
+To check that the gRPC server is running, try calling `scanner.Scanner/GetInfo`, for example with `grpcurl`:
+
+```bash
+$ grpcurl -plaintext '127.0.0.1:8231' scanner.Scanner/GetInfo
+```
+
+The response should look like:
+
+```
+{
+  "minSaplingBirthdayHeight": 419200
+}
+```
+
+An example request to the `Scan` method with `grpcurl` would look like: 
+
+```bash
+grpcurl -plaintext -d '{ "keys": { "key": ["sapling_extended_full_viewing_key"] } }' '127.0.0.1:8231' scanner.Scanner/Scan
```

Comment:
```
alfredo@spaceship:~/zebra$ ./grpcurl -plaintext -d '{ "keys": { "key": ["sapling_extended_full_viewing_key"] } }' '127.0.0.1:8231' scanner.Scanner/Scan
Error invoking method "scanner.Scanner/Scan": service "scanner.Scanner" does not include a method named "Scan"
alfredo@spaceship:~/zebra$ 
```

Reply:
This worked for me just now after rebasing on main:

```console

grpcurl -plaintext -d '{ "keys": { "key": ["sapling_extended_full_viewing_key"] } }' '127.0.0.1:8231' scanner.Scanner/Scan
ERROR:
  Code: Unknown
  Message: scan service returned error: channel closed
```

Reply:
I think the problem might be if the state is below the sapling activation or similar. When i use the command i get a hang (this is more than 10 or 30 seconds, it seems to be something else).

Reply:
Ok, i have more details in the hangs. If we are below sapling activation the grpc methods that require any state will hang. I can see this messages in zebrad logs:

```
zebra_scan::service::scan_task::scan: scanner is waiting for Sapling activation. Current tip: 106951, Sapling activation: 419200
```

So we need to return an error from the grpc call if we are below sapling activation height. We can do this in another PR.

Reply:
Should we add an error about the invalid key used?

```bash
alfredo@spaceship:~/zebra$ ./grpcurl -plaintext -d '{ "keys": { "key": ["sapling_extended_full_viewing_key"] } }' '127.0.0.1:8231' scanner.Scanner/Scan
ERROR:
  Code: Unknown
  Message: scan service returned error: channel closed
alfredo@spaceship:~/zebra$
```

Why this is not working? the error doesn't tell us too much.
```
alfredo@spaceship:~/zebra$ ./grpcurl -plaintext -d '{ "keys": { "key": ["zxviews1q0duytgcqqqqpqre26wkl45gvwwwd706xw608hucmvfalr759ejwf7qshjf5r9aa7323zulvz6plhttp5mltqcgs9t039cx2d09mgq05ts63n8u35hyv6h9nc9ctqqtue2u7cer2mqegunuulq2luhq3ywjcz35yyljewa4mgkgjzyfwh6fr6jd0dzd44ghk0nxdv2hnv4j5nxfwv24rwdmgllhe0p8568sgqt9ckt02v2kxf5ahtql6s0ltjpkckw8gtymxtxuu9gcr0swvz"] } }' '127.0.0.1:8231' scanner.Scanner/Scan
ERROR:
  Code: Unknown
  Message: scan service returned error: channel closed
alfredo@spaceship:~/zebra$ 
```

Reply:
Yea, I also think the hang is not a blocker for this PR. It would be good to get it merged before the release.

Reply:
I opened a separated ticket to work on those (https://github.com/ZcashFoundation/zebra/issues/8316). Feel free to close all conversations and merge.